enhancements/ingress/optional-ingress-hypershift.md

```diff
@@ -0,0 +1,323 @@
+
+### Allow standalone OpenShift to be broken if the ingress is disabled
+
+This approach implies the implementation of the ingress capability the way it's described in [how to implement a new capability](/enhancements/installer/component-selection.md#how-to-implement-a-new-capability).
+But unlike this enhancement, it doesn't have a goal to tolerate the absence of the ingress operator.
+That is, anything which can break during or after the cluster installation is allowed to break.
+This approach has a clear downside of the postponed failure which questions the correctness of the capability implementation from the user's point of view.
+The upside is the simpler implementation and the absence of the hidden magic going on under the covers. The magic which a user might accidentally or deliberately circumvent by, for example, rendering the cluster resources and then modifying the `ClusterVersion` object prior to creating the cluster.
+
+### Forbid the standalone installation without the ingress capability
+
+This approach is an alternative to the proposed implementation of the "always enabled" capabilities in the cluster version operator.
+The idea is to make the installer forbid the installation without the ingress capability.
+This could be achieved either via a validation error [after](https://github.com/openshift/installer/blob/c47f0d0d112aacc52d0ee2f0cf413f43905f56a1/pkg/types/validation/installconfig.go#L59) the unmarshaling of the install config
+or via the implicit enabling of the capability [before](https://github.com/openshift/installer/blob/c47f0d0d112aacc52d0ee2f0cf413f43905f56a1/pkg/types/defaults/installconfig.go#L32) it's passed to the cluster version operator.      
+
+While having similarly poor level of the user experience the implementation in the cluster version operator has some advantages over this approach:
+- it's a better fit for different OpenShift form factors: standalone, SNO, etc.
+- the logic for the capabilities to be always enabled pairs well with [the existing implicit enabling](https://github.com/openshift/cluster-version-operator/blob/2fd54d18de83b617d9d30c28cfee6383e430a102/pkg/cvo/status.go#L161-L165), so both are better to belong to the same repository.
+
+### Alternative to the cluster capability to disable the ingress operator
+
+Due to the lack of understanding of how HyperShift uses the cluster version operator to manage the ingress operator, we thought that the capability would serve only the informative purpose for the other OpenShift components like the console operator.
+That brought the idea of giving up with the capability in favor of some other cluster resource (existing or not) which could fulfill the need of informing other OpenShift components.
+Potentially that could save us from the difficult design and implementation decisions which the cluster capability implies. [The hypershift chapter](#hypershift) explains in details why the capability should be preferred over any other API.
```

Comment:
This section confuses me.  After reading it a few times, I think you're saying this:

* If we added a custom API, then we would need to do these two things:
  1. Define a new "turn ingress off" API.  
  2. Modify HyperShift not to install the ingress operator when that API were used.  
* However, the HyperShift team intends to add capabilities support to HyperShift, which means we only need to do these two things:
  1. Define a new ingress capability, which is easier than defining a custom API.
  2. Annotate the ingress operator's CVO manifests with the new capability, which is easier than modifying HyperShift.  

Do I understand correctly?

Reply:
> Do I understand correctly?

Yes.